pyro/infer/predictive.py

```diff
@@ -450,3 +450,181 @@ def forward(self, *args, **kwargs):
             guide_log_prob=guide_log_prob,
             model_log_prob=model_log_prob,
         )
+
+
+class MHResampler(torch.nn.Module):
+    r"""
+    Resampler for weighed samples that generates equally weighed samples from the distribution
```

Comment:
Gosh I guess I missed this in the previous review, but shouldn't we use the standard term `weighted` rather than `weighed`? I interpret `weighted` as "individually annotated by weights", whereas `weighed` sounds like the state of a slab of fish just before the fishmonger stamps on its price tag.

Reply:
I think `weighed` is good if you look at it as derived from the verb to `weigh`, which is to ascertain the weight or importance of something,, so you have `weighed` samples, i.e. samples for which their weights have been ascertained.